### PR TITLE
Fix to make it consistent with numpy.split and cupy.split

### PR DESCRIPTION
**Pull request by ken-nakanishi**
Fix to make it consistent with numpy.split and cupy.split

before
```
>>> import numpy as np
>>> import cupy as cp
>>> import chainer.functions as F
>>> np.split(np.arange(4), indices_or_sections=[0,1,1], axis=0)
[array([], dtype=int64), array([0]), array([], dtype=int64), array([1, 2, 3])]
>>> cp.split(cp.arange(4), indices_or_sections=[0,1,1], axis=0)
[array([], dtype=int64), array([0]), array([], dtype=int64), array([1, 2, 3])]
>>> F.split_axis(np.arange(4), indices_or_sections=[0,1,1], axis=0)
(variable([]), variable([0]), variable([]), variable([1, 2, 3]))
>>> np.split(np.arange(4), indices_or_sections=[0,1,1,4], axis=0)
[array([], dtype=int64), array([0]), array([], dtype=int64), array([1, 2, 3]), array([], dtype=int64)]
>>> cp.split(cp.arange(4), indices_or_sections=[0,1,1,4], axis=0)
[array([], dtype=int64), array([0]), array([], dtype=int64), array([1, 2, 3]), array([], dtype=int64)]
>>> F.split_axis(np.arange(4), indices_or_sections=[0,1,1,4], axis=0)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/***/chainer/functions/array/split_axis.py", line 92, in split_axis
    res = SplitAxis(indices_or_sections, axis).apply((x,))
  File "/***/chainer/function_node.py", line 230, in apply
    self._check_data_type_forward(in_data)
  File "/***/chainer/function_node.py", line 298, in _check_data_type_forward
    self.check_type_forward(in_type)
  File "/***/chainer/functions/array/split_axis.py", line 37, in check_type_forward
    type_check.expect(in_types[0].shape[self.axis] > max_index)
  File "/***/chainer/utils/type_check.py", line 524, in expect
    expr.expect()
  File "/***/chainer/utils/type_check.py", line 482, in expect
    '{0} {1} {2}'.format(left, self.inv, right))
chainer.utils.type_check.InvalidType:
Invalid operation is performed in: SplitAxis (Forward)

Expect: in_types[0].shape[0] > max_index
Actual: 4 <= 4
```

after
```
>>> import numpy as np
>>> import chainer.functions as F
>>> F.split_axis(np.arange(4), indices_or_sections=[0,1,1,4], axis=0)
(variable([]), variable([0]), variable([]), variable([1, 2, 3]), variable([]))
```

### CODE REVIEW

**Comment by mitmul** [comment]
Jenkins, test this please.

**Comment by niboshi** [comment]
Could you add a test case in `tests/chainer_tests/functions_tests/array_tests/test_split_axis.py`?

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by niboshi** [comment]
jenkins, test this please.

**Comment by niboshi** [comment]
This PR fixes #4333.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
Cloud you resolve conflicts?

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by okuta** [comment]
@ken-nakanishi Do you have time to fix this PR?

**Comment by ken-nakanishi** [comment]
Sorry for the late notice. I will start this task immediately.

File: tests/chainer_tests/functions_tests/array_tests/test_split_axis.py [hunk 1]
```diff
@@ -109,6 +109,9 @@ def inject_backend_tests(method_names):
         {'dtype': numpy.float64},
     ],
 ))
+@unittest.skipIf(numpy.lib.NumpyVersion(numpy.__version__) <= '1.9.3',
```
Comment (okuta) [inline tests/chainer_tests/functions_tests/array_tests/test_split_axis.py]: You can use more simple method.
`testing.with_requires('numpy>=1.10')`

**Comment by okuta** [comment]
jenkins, test this please.

**Comment by ken-nakanishi** [comment]
Noted with thanks. Now I fixed it.

**Comment by ken-nakanishi** [comment]
Noted with thanks. Now I fixed it.

**Comment by okuta** [comment]
Thanks!

jenkins, test this please.

**Comment by okuta** [comment]
LGTM!

**Pull request merged**